esp-hal/src/gpio/mod.rs

```diff
@@ -61,6 +61,25 @@
 //!
 //! See the [Blinky][crate#blinky] section of the crate documentation.
 //!
+//! ### Type-erased pins usage
+//! ```rust, no_run
+#![doc = crate::before_snippet!()]
+//! # use esp_hal::gpio::{Io, Level, Input, Pull};
+//! # use esp_hal::delay::Delay;
+//! let mut button = peripherals.GPIO5.degrade();
```

Comment:
We shouldn't write examples that showcase something non-valuable. The example should give an idea where a certain function may be useful - and calling `degrade` just to pass the pin to a driver is just noise.